torch/nn/functional.py

```diff
@@ -592,27 +594,47 @@ def adaptive_avg_pool3d(input, output_size):
 
 # Activation functions
 def dropout(input, p=0.5, training=False, inplace=False):
-    return _functions.dropout.Dropout.apply(input, p, training, inplace)
+    if p < 0 or p > 1:
```

Comment:
Why isn't this checked inside the C++ operator itself?

Reply:
Because there's no way to raise a ValueError from ATen, but I wanted to keep the interface consistent (we do have a test for this). I can remove this and update the test to throw a RuntimeError if you'd prefer that.